### PR TITLE
`matplotlib` (via `seaborn`) support for `coefplot` and `iplot`

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/s3alfisc/pyfixest/issues/378

### CODE REVIEW

**Comment by juanitorduz** [comment]
@s3alfisc I think this one is ready for a first round of review. 

I noticed these plotting functions do not use the `alpha` parameter, even though it is in the function signature. I suggest we do not change this here (as I am unfamiliar with lets_plot) but rather create an issue so the PRs are scoped and easier to review. WDYT?

File: pyfixest/report/visualize.py [hunk 1]
```diff
@@ -235,7 +258,17 @@ def coefplot(
...
 
-def _coefplot(
+def _coefplot(plot_backend, **plot_kwargs):
+    """Coefplot function that dispatches to the correct plotting backend."""
+    if plot_backend == "lets_plot":
+        return _coefplot_lets_plot(**plot_kwargs)
+    elif plot_backend == "matplotlib":
+        return _coefplot_matplotlib(**plot_kwargs)
+    else:
+        raise ValueError("plot_backend must be either 'lets_plot' or 'matplotlib'.")
```
Comment (s3alfisc) [inline pyfixest/report/visualize.py:268]: Here we could add a very simple test that triggers this error to `test_errors.py`. I can do this tomorrow morning.

File: pyfixest/report/visualize.py [hunk 2]
```diff
@@ -223,7 +245,8 @@ def coefplot(
         idxs = df.index
     df = df.loc[idxs, :].reset_index()
 
-    return _coefplot(
+    _coefplot(
```
Comment (s3alfisc) [inline pyfixest/report/visualize.py:248]: The `return` statement is missing, same as above.

File: pyfixest/did/lpdid.py [hunk 3]
```diff
@@ -163,7 +166,8 @@ def iplot(
         df = self._coeftable
         df["fml"] = "lpdid"
 
-        return _coefplot(
+        _coefplot(
```
Comment (s3alfisc) [inline pyfixest/did/lpdid.py:169]: The return statement is missing. This leads to the error when building the docs in the CI: 

```python
import pandas as pd
from pyfixest.did.estimation import lpdid

url = "https://raw.githubusercontent.com/s3alfisc/pyfixest/master/pyfixest/did/data/df_het.csv"
df_het = pd.read_csv(url)

fit = lpdid(
    df_het,
    yname="dep_var",
    idname="unit",
    tname="year",
    gname="g",
    vcov={"CRV1": "state"},
    pre_window=-20,
    post_window=20,
    att=False
)

fit.tidy().head()
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
------------------


---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[1], line 20
      7 fit = lpdid(
      8     df_het,
      9     yname="dep_var",
   (...)
     16     att=False
     17 )
     19 fit.tidy().head()
---> 20 fit.iplot(figsize= [1200, 400], coord_flip=False).show()

AttributeError: 'NoneType' object has no attribute 'show'
```

**Comment by s3alfisc** [comment]
Yes, let's have a separate issue for the `alpha` parameter. To enable it, it would likely be best to add an argument `alpha` to [Feols.tidy()](https://github.com/s3alfisc/pyfixest/blob/af27baf87f8b2d78f17c11e55be43cb07a480c34/pyfixest/estimation/feols_.py#L1417), in which we would then have to call `se = self.confint(alpha = alpha)` instead of assigning `_se = self._se`. On of these "smaller" rough edges I have not yet managed to polish... Generally I also need to review if the package consistently uses `alpha=0.05` vs `alpha=0.95` when setting CI levels. 

I have added a test that triggers the `ValueError("plot_backend must be either 'lets_plot' or 'matplotlib'.")`.

Btw, the more detailed pytest error / debug messages are awesome! 😄

File: tests/test_errors.py [hunk 4]
```diff
@@ -333,6 +333,7 @@ def test_i_error():
     with pytest.raises(FactorEvaluationError):
         feols("Y ~ i(f1, X1, ref=a)", data)
 
+
```
Comment (s3alfisc) [inline tests/test_errors.py:336]: Sorry, pre-commit does not seem to be set up correctly on my local machine.

File: pyfixest/report/visualize.py (hunk 1 above)
Comment (juanitorduz) [reply pyfixest/report/visualize.py:268]: Yes! I was planning on adding it after the initial feedback 😄

File: tests/test_errors.py [hunk 5]
```diff
@@ -332,3 +332,10 @@ def test_i_error():
 
     with pytest.raises(FactorEvaluationError):
         feols("Y ~ i(f1, X1, ref=a)", data)
+
+
+def visualization_error():
```
Comment (juanitorduz) [inline tests/test_errors.py:337]: The function name has to start with test_

File: tests/test_errors.py (hunk 4 above)
Comment (juanitorduz) [reply tests/test_errors.py:336]: all good! :)

**Comment by juanitorduz** [comment]
I will add similar tests to pilot.

File: pyfixest/did/lpdid.py (hunk 3 above)
Comment (juanitorduz) [reply pyfixest/did/lpdid.py]: Fixed 👍

File: pyfixest/report/visualize.py (hunk 2 above)
Comment (juanitorduz) [reply pyfixest/report/visualize.py]: fixed

**Comment by s3alfisc** [comment]
This looks excellent, thanks so much Juan! I will merge it as soon as the CI tests pass =)

**Review by s3alfisc — APPROVED** [review]

**Pull request merged**